### PR TITLE
translated using typescript page

### PR DESCRIPTION
**Pull request by shajahanshaik915**

### CODE REVIEW

**Comment by shajahanshaik915** [comment]
@srikanth-kandi Could you please review my PR? Thank you!

File: src/content/learn/typescript.md [hunk 1]
```diff
@@ -424,7 +422,7 @@ interface ModalRendererProps {
 }
 ```
 
-This is a very broad definition of children. The second is to use the `React.ReactElement` type, which is only JSX elements and not JavaScript primitives like strings or numbers:
+ఇది children యొక్క విస్తృత నిర్వచనం. రెండవది `React.ReactElement` type ను ఉపయోగించడం, ఇది కేవలం JSX elements మాత్రమే, JavaScript primitives వంటి strings లేదా numbers కాదు:
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:425]:
```suggestion
ఇది చైల్డ్ ఎలిమెంట్ యొక్క బ్రాడ్ డెఫినిషన్. రెండవది `React.ReactElement` టైప్ ను ఉపయోగించడం, ఇది కేవలం JSX ఎలెమెంట్స్ మాత్రమే, JavaScript primitives వంటి strings లేదా numbers కాదు:
```

File: src/content/learn/typescript.md [hunk 2]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
 }
 ```
 
-Note, that you cannot use TypeScript to describe that the children are a certain type of JSX elements, so you cannot use the type-system to describe a component which only accepts `<li>` children. 
+గమనించండి, మీరు TypeScript ను children ఒక నిర్దిష్ట రకం JSX elements అని వివరించడానికి ఉపయోగించలేరు, కాబట్టి కేవలం `<li>` children మాత్రమే అంగీకరించే ఒక కంపోనెంట్ ను type-system ద్వారా వివరించలేరు.
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:434]:
```suggestion
గమనించండి, మీరు TypeScript ను చైల్డ్ ఎలిమెంట్ ఒక నిర్దిష్ట టైప్ JSX ఎలిమెంట్స్ ని వివరించడానికి ఉపయోగించలేరు, కాబట్టి కేవలం `<li>` చైల్డ్ ఎలిమెంట్ మాత్రమే అంగీకరించే ఒక కంపోనెంట్ ను టైప్-సిస్టమ్‌ ద్వారా వివరించలేరు.
```

File: src/content/learn/typescript.md [hunk 3]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
+`React.ReactNode` మరియు `React.ReactElement` రెండింటి ఉదాహరణను type-checker తో సహా [ఈ TypeScript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgIilQ3wChSB6CxYmAOmXRgDkIATJOdNJMGAZzgwAFpxAR+8YADswAVwGkZMJFEzpOjDKw4AFHGEEBvUnDhphwADZsi0gFw0mDWjqQBuUgF9yaCNMlENzgAXjgACjADfkctFnYkfQhDAEpQgD44AB42YAA3dKMo5P46C2tbJGkvLIpcgt9-QLi3AEEwMFCItJDMrPTTbIQ3dKywdIB5aU4kKyQQKpha8drhhIGzLLWODbNs3b3s8YAxKBQAcwXpAThMaGWDvbH0gFloGbmrgQfBzYpd1YjQZbEYARkB6zMwO2SHSAAlZlYIBCdtCRkZpHIrFYahQYQD8UYYFA5EhcfjyGYqHAXnJAsIUHlOOUbHYhMIIHJzsI0Qk4P9SLUBuRqXEXEwAKKfRZcNA8PiCfxWACecAAUgBlAAacFm80W-CU11U6h4TgwUv11yShjgJjMLMqDnN9Dilq+nh8pD8AXgCHdMrCkWisVoAet0R6fXqhWKhjKllZVVxMcavpd4Zg7U6Qaj+2hmdG4zeRF10uu-Aeq0LBfLMEe-V+T2L7zLVu+FBWLdLeq+lc7DYFf39deFVOotMCACNOCh1dq219a+30uC8YWoZsRyuEdjkevR8uvoVMdjyTWt4WiSSydXD4NqZP4AymeZE072ZzuUeZQKheQgA) లో చూడవచ్చు.
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:436]:
```suggestion
`React.ReactNode` మరియు `React.ReactElement` రెండింటి ఉదాహరణను టైప్-చెక్కర్ తో సహా [ఈ TypeScript ప్లేగ్రౌండ్](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgIilQ3wChSB6CxYmAOmXRgDkIATJOdNJMGAZzgwAFpxAR+8YADswAVwGkZMJFEzpOjDKw4AFHGEEBvUnDhphwADZsi0gFw0mDWjqQBuUgF9yaCNMlENzgAXjgACjADfkctFnYkfQhDAEpQgD44AB42YAA3dKMo5P46C2tbJGkvLIpcgt9-QLi3AEEwMFCItJDMrPTTbIQ3dKywdIB5aU4kKyQQKpha8drhhIGzLLWODbNs3b3s8YAxKBQAcwXpAThMaGWDvbH0gFloGbmrgQfBzYpd1YjQZbEYARkB6zMwO2SHSAAlZlYIBCdtCRkZpHIrFYahQYQD8UYYFA5EhcfjyGYqHAXnJAsIUHlOOUbHYhMIIHJzsI0Qk4P9SLUBuRqXEXEwAKKfRZcNA8PiCfxWACecAAUgBlAAacFm80W-CU11U6h4TgwUv11yShjgJjMLMqDnN9Dilq+nh8pD8AXgCHdMrCkWisVoAet0R6fXqhWKhjKllZVVxMcavpd4Zg7U6Qaj+2hmdG4zeRF10uu-Aeq0LBfLMEe-V+T2L7zLVu+FBWLdLeq+lc7DYFf39deFVOotMCACNOCh1dq219a+30uC8YWoZsRyuEdjkevR8uvoVMdjyTWt4WiSSydXD4NqZP4AymeZE072ZzuUeZQKheQgA) లో చూడవచ్చు.
```

File: src/content/learn/typescript.md [hunk 4]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
 
 ### Style Props {/*typing-style-props*/}
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:438]:
```suggestion
### స్టైల్ props {/*typing-style-props*/}
```

File: src/content/learn/typescript.md [hunk 5]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
 
-When using inline styles in React, you can use `React.CSSProperties` to describe the object passed to the `style` prop. This type is a union of all the possible CSS properties, and is a good way to ensure you are passing valid CSS properties to the `style` prop, and to get auto-complete in your editor.
+React లో inline styles ఉపయోగిస్తున్నప్పుడు, `style` prop కు పంపబడే object ను వర్ణించడానికి మీరు `React.CSSProperties` ను ఉపయోగించవచ్చు. ఈ type అన్ని సంభావ్య CSS properties యొక్క యూనియన్, మరియు ఇది మీరు `style` prop కు సరైన CSS properties పంపిస్తున్నారని నిర్ధారించడానికి మరియు మీ ఎడిటర్ లో auto-complete పొందడానికి మంచి మార్గం.
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:440]:
```suggestion
React లో ఇన్లైన్ స్టైల్స్ ఉపయోగిస్తున్నప్పుడు, `style` prop కు పంపబడే ఆబ్జెక్ట్ ను డిస్క్రైబ్ చేయడానికి మీరు `React.CSSProperties` ను ఉపయోగించవచ్చు. ఈ టైప్ అన్ని పాసిబుల్ CSS ప్రాపర్టీస్ యొక్క యూనియన్, మరియు ఇది మీరు `style` prop కు సరైన CSS ప్రాపర్టీస్ పంపిస్తున్నారని నిర్ధారించడానికి మరియు మీ ఎడిటర్ లో ఆటో-కంప్లీట్ పొందడానికి మంచి మార్గం.
```

File: src/content/learn/typescript.md [hunk 6]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
 
 ```ts
 interface MyComponentProps {
   style: React.CSSProperties;
 }
 ```
 
-## Further learning {/*further-learning*/}
+## మరింత నేర్చుకోవడానికి {/*further-learning*/}
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:448]:
```suggestion
## మరింత నేర్చుకోవడానికి రిసోర్సెస్ {/*further-learning*/}
```

File: src/content/learn/typescript.md [hunk 7]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
 
-This guide has covered the basics of using TypeScript with React, but there is a lot more to learn.
-Individual API pages on the docs may contain more in-depth documentation on how to use them with TypeScript.
+ఈ గైడ్ React తో TypeScript ఉపయోగించడం గురించి ప్రాథమికాలను కవర్ చేసింది, కానీ ఇంకా నేర్చుకోవడానికి చాలా ఉంది.
+Docs లోని వ్యక్తిగత API పేజీలు TypeScript తో వాటిని ఎలా ఉపయోగించాలో గురించి మరింత లోతైన డాక్యుమెంటేషన్ కలిగి ఉండవచ్చు.
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:451]:
```suggestion
ఈ గైడ్ React తో TypeScript ఉపయోగించడం గురించి బేసిక్స్ ను కవర్ చేసింది, కానీ ఇంకా నేర్చుకోవడానికి చాలా ఉంది.
డాక్స్ లోని ఇండివిడ్యుఅల్ API పేజీలలో TypeScript తో వాటిని ఎలా ఉపయోగించాలి అనే దాని గురించి మరింత ఇన్-డెప్త్ డాక్యుమెంటేషన్ కలిగి ఉండవచ్చు.
```

File: src/content/learn/typescript.md [hunk 8]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
 
-We recommend the following resources:
+మేము ఈ కింది వనరులను సిఫారసు చేస్తున్నాము:
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:453]:
```suggestion
మేము ఈ కింది రిసోర్సెస్ ను సిఫార్సు చేస్తున్నాము:
```

File: src/content/learn/typescript.md [hunk 9]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
 
- - [The TypeScript handbook](https://www.typescriptlang.org/docs/handbook/) is the official documentation for TypeScript, and covers most key language features.
+ - [The TypeScript handbook](https://www.typescriptlang.org/docs/handbook/) TypeScript కోసం అధికారిక డాక్యుమెంటేషన్, మరియు అధిక భాగం కీలక భాషా లక్షణాలను కవర్ చేస్తుంది.
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:455]:
```suggestion
 - [TypeScript హ్యాండ్‌బుక్](https://www.typescriptlang.org/docs/handbook/) TypeScript కోసం ఆఫిసిఅల్ డాక్యుమెంటేషన్, మరియు అధిక భాగం కీలక లాంగ్వేజ్ ఫీచర్లను కవర్ చేస్తుంది.
```

File: src/content/learn/typescript.md [hunk 10]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
 
- - [The TypeScript release notes](https://devblogs.microsoft.com/typescript/) cover new features in depth.
+ - [The TypeScript release notes](https://devblogs.microsoft.com/typescript/) కొత్త లక్షణాలను లోతుగా కవర్ చేస్తాయి.
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:457]:
```suggestion
 - [TypeScript రిలీజ్ నోట్స్](https://devblogs.microsoft.com/typescript/) కొత్త ఫీచర్లను ఇన్-డెప్త్ గా కవర్ చేస్తాయి.
```

File: src/content/learn/typescript.md [hunk 11]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
 
- - [React TypeScript Cheatsheet](https://react-typescript-cheatsheet.netlify.app/) is a community-maintained cheatsheet for using TypeScript with React, covering a lot of useful edge cases and providing more breadth than this document.
+ - [React TypeScript Cheatsheet](https://react-typescript-cheatsheet.netlify.app/) TypeScript తో React ఉపయోగించడానికి కమ్యూనిటీ-నిర్వహిత చీట్ షీట్, చాలా ఉపయోగకరమైన ఎడ్జ్ కేసులను కవర్ చేస్తూ ఈ డాక్యుమెంట్ కంటే మరింత విస్తృతతనాన్ని అందిస్తుంది.
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:459]:
```suggestion
 - [React TypeScript చీట్ షీట్](https://react-typescript-cheatsheet.netlify.app/) TypeScript తో React ఉపయోగించడానికి కమ్యూనిటీ-మైన్టైన్డ్ చీట్ షీట్, చాలా ఉపయోగకరమైన ఎడ్జ్ కేసులను కవర్ చేస్తూ ఈ డాక్యుమెంట్ కంటే మరింత విస్తృత తనాన్ని అందిస్తుంది.
```

File: src/content/learn/typescript.md [hunk 12]
```diff
@@ -433,31 +431,31 @@ interface ModalRendererProps {
...
 
- - [TypeScript Community Discord](https://discord.com/invite/typescript) is a great place to ask questions and get help with TypeScript and React issues.
+ - [TypeScript Community Discord](https://discord.com/invite/typescript) TypeScript మరియు React సమస్యలతో సహాయం కోసం ప్రశ్నలు అడగడానికి మరియు సహాయం పొందడానికి గొప్ప ప్రదేశం.
```
Comment (srikanth-kandi) [inline src/content/learn/typescript.md:461]:
```suggestion
 - [TypeScript కమ్యూనిటీ Discord](https://discord.com/invite/typescript) TypeScript మరియు React సమస్యలతో సహాయం కోసం ప్రశ్నలు అడగడానికి మరియు సహాయం పొందడానికి గొప్ప ప్రదేశం.
```

**Review by srikanth-kandi — CHANGES_REQUESTED** [review]

**Comment by srikanth-kandi** [comment]
Thanks @shajahanshaik915 for your contribution. I look forward to seeing more contributions from you 🤝

**Comment by shajahanshaik915** [comment]
I'm glad I could contribute, and I look forward to contribute more in the future 🤝.

**Pull request merged**